src/software/ai/navigator/obstacle/obstacle_factory.cpp

```diff
@@ -142,31 +141,55 @@ std::vector<ObstaclePtr> ObstacleFactory::createVelocityObstaclesFromTeam(
     return obstacles;
 }
 
-ObstaclePtr ObstacleFactory::createBallObstacle(const Point &ball_position)
+ObstaclePtr ObstacleFactory::createBallObstacle(const Point &ball_position) const
+{
+    return expandForRobotSize(Circle(ball_position, BALL_MAX_RADIUS_METERS + 0.06));
+}
+
+ObstaclePtr ObstacleFactory::createRobotObstacle(const Point &robot_position) const
 {
-    return createObstacle(
-        Circle(ball_position, BALL_MAX_RADIUS_METERS + 0.06 + shape_expansion_amount));
+    return expandForRobotSize(Circle(robot_position, ROBOT_MAX_RADIUS_METERS));
 }
 
-ObstaclePtr ObstacleFactory::createRobotObstacle(const Point &robot_position)
+ObstaclePtr ObstacleFactory::createObstacleFromRectangle(const Rectangle &rectangle) const
 {
-    return createObstacle(
-        Circle(robot_position, ROBOT_MAX_RADIUS_METERS + shape_expansion_amount));
+    return expandForRobotSize(Polygon(rectangle));
 }
 
-ObstaclePtr ObstacleFactory::createObstacleFromRectangle(const Rectangle &rectangle)
+ObstaclePtr ObstacleFactory::expandForRobotSize(const Circle &circle) const
 {
-    Rectangle rectangle_exp(rectangle);
-    rectangle_exp.inflate(shape_expansion_amount);
-    return createObstacle(rectangle_exp);
+    return std::make_shared<GeomObstacle<Circle>>(
+        Circle(circle.getOrigin(), circle.getRadius() + obstacle_expansion_amount));
 }
 
-ObstaclePtr ObstacleFactory::createObstacle(const Circle &circle)
+ObstaclePtr ObstacleFactory::expandForRobotSize(const Polygon &polygon) const
 {
-    return std::make_shared<GeomObstacle<Circle>>(GeomObstacle<Circle>(circle));
+    return std::make_shared<GeomObstacle<Polygon>>(
+        polygon.expand(obstacle_expansion_amount * Vector(-1, 0))
+            .expand(obstacle_expansion_amount * Vector(1, 0))
+            .expand(obstacle_expansion_amount * Vector(0, -1))
+            .expand(obstacle_expansion_amount * Vector(0, 1)));
 }
 
-ObstaclePtr ObstacleFactory::createObstacle(const Polygon &polygon)
+ObstaclePtr ObstacleFactory::expandForRobotSize(
+    const Rectangle &rectangle, const std::vector<Vector> &directions) const
 {
-    return std::make_shared<GeomObstacle<Polygon>>(GeomObstacle<Polygon>(polygon));
+    Rectangle retval = rectangle;
+    for (const auto &v : directions)
+    {
+        retval = retval.expand(obstacle_expansion_amount * v);
+    }
+    return std::make_shared<GeomObstacle<Polygon>>(retval);
+}
+
+ObstaclePtr ObstacleFactory::expandForInflatedRobotSize(
+    const Rectangle &rectangle, const std::vector<Vector> &directions) const
+{
+    double inflated_expansion_amount = obstacle_expansion_amount + 0.3;
```

Comment:
The 0.3 is the only difference between `expandForRobotSize` and `expandForInflatedRobotSize`. And its only used once on the defence area expansion. Why not just have an `expandRectangleObstacle` or something of that sorts and remove this function altogether?